app/app.go

```diff
@@ -576,7 +576,7 @@ func (app *App) EndBlocker(ctx sdk.Context, req abci.RequestEndBlock) abci.Respo
 	// NOTE: this is a specific feature for upgrading to v2 as v3 and onward is expected
 	// to be coordinated through the signalling protocol
 	if app.UpgradeKeeper.ShouldUpgrade(req.Height) {
-		app.SetProtocolVersion(v2.Version)
+		app.SetAppVersion(ctx, v2.Version)
```

Comment:
[Optional idea and not directly related to this PR] Wouldn't it be easier, in terms of future refactoring, to retrieve the app version based on the upgrade height rather than assuming that we know the next version is v2? A simple approach could involve using a mapping that associates the upgrade height with all the relevant configurations, including the app version. Totally up to you though.

Reply:
The v2 upgrade is a one off event and thus hardcoded this way. Future upgrades will be done more dynamically through a signalling mechanism that I am working on